src/dbnode/storage/index/results.go

```diff
@@ -128,29 +120,29 @@ func (r *results) addDocumentsBatchWithLock(batch []doc.Document) error {
 	return nil
 }
 
-func (r *results) addDocumentWithLock(d doc.Document) (bool, int, error) {
-	if len(d.ID) == 0 {
-		return false, r.resultsMap.Len(), errUnableToAddResultMissingID
+func (r *results) addDocumentWithLock(e doc.EncodedDocument) (bool, int, error) {
+	id, err := docs.ReadEncodedDocumentID(e)
+	if err != nil {
+		return false, r.resultsMap.Len(), err
 	}
 
-	// NB: can cast the []byte -> ident.ID to avoid an alloc
-	// before we're sure we need it.
-	tsID := ident.BytesID(d.ID)
```

Comment:
Hm I wonder if we still want to do this and have the key be `ident.BytesID`? I don't really follow what this comment is explaining though.